plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/data/AltibaseGeometryValueHandler.java

```diff
@@ -0,0 +1,179 @@
+                return new DBGeometry((Geometry) object);
+            } else {
+                /*
+                 * EWKT: 32,000 is maximum length of return value.
+                 * SRID=xxxx;MULTIPOLYGON (((199....
+                 */
+
+                Geometry geom = null;
+
+                String[] content = ((String) object).split(";", 2);
+
+                // start with SRID=xxxx;
+                if (content.length == 2) {
+
+                    // get srid
+                    content[0] = content[0].replaceAll("SRID=", "");
+                    try {
+                        srid = Integer.parseInt(content[0]);
+                    } catch (NumberFormatException e) {
+                        log.warn("Failed to parse SRID: " + e);
+                    }
+
+                    // geom from WKT
+                    geom = new WKTReader().read(content[1]);
+
+                } else if (content.length == 1) {
+                    // No SRID, just in case.
+                    geom = new WKTReader().read(content[0]);
+                }
+
+                dbGeometry = new DBGeometry(geom);
+                dbGeometry.setSRID(srid);
+            }
+        } catch (ParseException e) {
+            log.warn("Failed to parse objedt: " 
+                    + ((object != null) ? object.toString() : "NULL") 
+                    + AltibaseConstants.NEW_LINE 
+                    + e.getLocalizedMessage());
+        }
+
+        return dbGeometry;
+    }
+
+    @Override
+    protected void bindParameter(JDBCSession session, JDBCPreparedStatement statement, 
+            DBSTypedObject paramType, int paramIndex, Object value) throws DBCException, SQLException {
+
+        int srid = 0;
+        DBSTypedObject attribute = null;
+        Object rawValue = null;
+        
+        if (paramType instanceof DBDAttributeBinding) {
+            attribute = ((DBDAttributeBinding) paramType).getAttribute();
+        } else {
+            attribute = paramType;
+        }
+        
+        if (value instanceof DBGeometry) {
+            srid = ((DBGeometry) value).getSRID();
+            rawValue = ((DBGeometry) value).getRawValue();
+        } else {
+            rawValue = value;
+        }
+        
+        if (srid == 0 && attribute instanceof GisAttribute) {
+            srid = ((GisAttribute) attribute).getAttributeGeometrySRID(session.getProgressMonitor());
+        }
+        
+        if (rawValue == null) {
+            statement.setNull(paramIndex, attribute.getTypeID());
+        } else if (rawValue instanceof Geometry) {
+            if (((Geometry) rawValue).getSRID() == 0) {
+                ((Geometry) rawValue).setSRID(srid);
+            }
+            // format: GEOMFROMTEXT(?, SRID)
+            statement.setString(paramIndex, new WKTWriter(4).write((Geometry) rawValue));
+
+        } else {
+            String strValue = rawValue.toString();
+            if (srid != 0 && !strValue.startsWith("SRID=")) {
```

Comment:
Please convert "SRID=" into constant.

Reply:
Okay, it'll be in AltibaseConstants. Thanks :)